test/orange/orangeinp/ConvexRegion.test.cc

```diff
@@ -371,6 +371,145 @@ TEST_F(EllipsoidTest, standard)
+    EXPECT_FALSE(result.interior) << result.interior;
+
+    // TODO: check why this is not lower=(-,-,-), upper=(+,+,+)
+    EXPECT_VEC_SOFT_EQ((Real3{1, 1, -3}), result.exterior.lower());
+    EXPECT_VEC_SOFT_EQ((Real3{-1, -1, 3}), result.exterior.upper());
+}
+
+TEST_F(GenTrapTest, trd)
+{
+    auto result = this->test(GenTrap(3,
+                                     {{-1, -1}, {-1, 1}, {1, 1}, {1, -1}},
+                                     {{-2, -2}, {-2, 2}, {2, 2}, {2, -2}}));
+
+    static char const expected_node[] = "all(+0, -1, -2, +3, +4, -5)";
+    static char const* const expected_surfaces[]
+        = {"Plane: z=-3",
+           "Plane: z=3",
+           "Plane: n={0.98639,-0,0.1644}, d=-1.4796",
+           "Plane: n={0,0.98639,-0.1644}, d=1.4796",
+           "Plane: n={0.98639,0,-0.1644}, d=1.4796",
+           "Plane: n={0,0.98639,0.1644}, d=-1.4796"};
+
+    EXPECT_EQ(expected_node, result.node);
+    EXPECT_VEC_EQ(expected_surfaces, result.surfaces);
+    EXPECT_FALSE(result.interior) << result.interior;
+    EXPECT_VEC_SOFT_EQ((Real3{-inf, -inf, -3}), result.exterior.lower());
+    EXPECT_VEC_SOFT_EQ((Real3{inf, inf, 3}), result.exterior.upper());
+}
+
+TEST_F(GenTrapTest, ppiped)
+{
+    auto result = this->test(GenTrap(4,
+                                     {{-2, -2}, {-2, 0}, {0, 0}, {0, -2}},
+                                     {{0, 0}, {0, 2}, {2, 2}, {2, 0}}));
+
+    static char const expected_node[] = "all(+0, -1, -2, +3, +4, -5)";
+    static char const* const expected_surfaces[]
+        = {"Plane: z=-4",
+           "Plane: z=4",
+           "Plane: n={0.97014,0,-0.24254}, d=-0.97014",
+           "Plane: n={0,0.97014,-0.24254}, d=0.97014",
+           "Plane: n={0.97014,0,-0.24254}, d=0.97014",
+           "Plane: n={0,0.97014,-0.24254}, d=-0.97014"};
+
+    EXPECT_EQ(expected_node, result.node);
+    EXPECT_VEC_EQ(expected_surfaces, result.surfaces);
+    EXPECT_FALSE(result.interior) << result.interior;
+    EXPECT_VEC_SOFT_EQ((Real3{-inf, -inf, -4}), result.exterior.lower());
+    EXPECT_VEC_SOFT_EQ((Real3{inf, inf, 4}), result.exterior.upper());
+}
+
+TEST_F(GenTrapTest, triang_prism)
+{
+    auto result = this->test(
+        GenTrap(3, {{-1, -1}, {2, 0}, {-1, 1}}, {{-1, -1}, {2, 0}, {-1, 1}}));
+
+    static char const expected_node[] = "all(+0, -1, -2, -3, +4)";
+    static char const* const expected_surfaces[]
+        = {"Plane: z=-3",
+           "Plane: z=3",
+           "Plane: n={0.31623,-0.94868,0}, d=0.63246",
+           "Plane: n={0.31623,0.94868,-0}, d=0.63246",
+           "Plane: x=-1"};
+
+    EXPECT_EQ(expected_node, result.node);
+    EXPECT_VEC_EQ(expected_surfaces, result.surfaces);
+    EXPECT_FALSE(result.interior) << result.interior;
+    EXPECT_VEC_SOFT_EQ((Real3{-1, -inf, -3}), result.exterior.lower());
+    EXPECT_VEC_SOFT_EQ((Real3{inf, inf, 3}), result.exterior.upper());
+}
+
+// TODO: this should be valid
+/*TEST_F(GenTrapTest, pentahedron)
```

Comment:
You can use the `DISABLED_` prefix to make sure that the code compiles: and you can test it by running with special googletest flags.
```suggestion
// TODO: not yet implemented
TEST_F(GenTrapTest, DISABLED_pentahedron)
```